### PR TITLE
pgd_attack: Reconcile C&W loss with @carlini's implementation

### PR DESCRIPTION
**Pull request by wh0**
Upstream implementation: https://github.com/carlini/nn_robust_attacks/blob/master/l2_attack.py#L90

This subtraction helps select the 'other' class when the logits are negative.

### CODE REVIEW

**Comment by dtsip** [comment]
Agreed, good catch. It turns out that the performance of the attack is not affected by this (at least on our public adversarially trained model), probably because the second largest logit is positive in most (if not all) of the cases.
